sdk/keyvault/Azure.Security.KeyVault.Keys/src/Cryptography/SignatureAlgorithmTranslator.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Security.Cryptography;
+
+namespace Azure.Security.KeyVault.Keys.Cryptography
+{
+    internal static class SignatureAlgorithmTranslator
+    {
+        public static SignatureAlgorithm SignatureAlgorithmToJwsAlgId(KeyVaultSignatureAlgorithm signatureAlgorithm, HashAlgorithmName hashAlgorithmName)
+        {
+            if (signatureAlgorithm == KeyVaultSignatureAlgorithm.RSAPkcs15)
+            {
+                if (hashAlgorithmName == HashAlgorithmName.SHA1)
+                    return new SignatureAlgorithm("RSNULL");
```

Comment:
Only support hash algorithms that Key Vault does in these scenarios, which does not include SHA1.